### PR TITLE
fix: build index for bucket after creation

### LINKED ISSUE #526: [Bug] Reading newly created bucket(after commit) will return `Bucket not found`
**Describe the bug**
If We create a bucket and try to read it immediately (Get), We'll get errors that the bucket is not found. I analysed and debugged the code and found that, We only build indexes when committing if there are some pendingWrites only.  Therefore, We won't build tx.db.buildBTreeIdx after bucket creation. 

**To Reproduce**
Steps to reproduce the behavior(Be specific!):
1. The below can reproduce this bug easily.

Give sample code if you can.
```go
package main

import (
	"fmt"
	"io/ioutil"
	"os"

	"github.com/nutsdb/nutsdb"
)

var (
	db     *nutsdb.DB
	bucket string
)

func init() {
	fileDir := "/tmp/nutsdb_example"

	files, _ := ioutil.ReadDir(fileDir)
	for _, f := range files {
		name := f.Name()
		if name != "" {
			fmt.Println(fileDir + "/" + name)
			err := os.RemoveAll(fileDir + "/" + name)
			if err != nil {
				panic(err)
			}
		}
	}
	db, _ = nutsdb.Open(
		nutsdb.DefaultOptions,
		nutsdb.WithDir(fileDir),
		nutsdb.WithSegmentSize(1024*1024), // 1MB
	)
	bucket = "bucketForString"
}

func main() {
	err := db.Update(func(tx *nutsdb.Tx) error {
		err := tx.NewBucket(nutsdb.DataStructureBTree, bucket)
		return err
	})
	if err != nil {
		fmt.Println("Errror in creating bucket")
		return
	}
	err = db.View(func(tx *nutsdb.Tx) error {
		// tx.db.Index.bTree.exist(bucketId) this check will fail 
		_, err := tx.Get(bucket, []byte("key"))
		return err
	})
	if err != nil {
		fmt.Println(err)
		return
	}
}
```

**Expected behavior**
We should not get `Bucket Not found " errors after creating bucket successfully. After creating a bucket, We may have to build indexes.

**What actually happens**
A clear and concise description of what actually happens.

https://github.com/nutsdb/nutsdb/blob/master/tx.go#L277 Here, We only build indexes after pending writes. So If We create a bucket and insert some entries into it, We won't get this error.

**Screenshots**

**please complete the following information :**
 - OS:  Mac OS
 - NutsDB Version 1.0.1

**Additional context**

### PR DESCRIPTION
**Pull request by ksankeerth**
Hi Team,

This fixes #526 .  I'm aware that We cannot use the bucket created in the same transaction Tx. But this is about the visibility of the bucket in another transaction Tx after bucket creation was successful. 

Please see the example I've provided in #526 

Thanks,
Sankeerthan

### CODE REVIEW

File: tx.go [hunk 1]
```diff
@@ -738,6 +745,31 @@ func (tx *Tx) DeleteBucketInIndex() error {
 	return nil
 }
 
+// buildBucketInIndex build indexes for newly created buckets
+// This method can be combined with DeleteBucketInIndex. As DeleteBucketInIndex is a public API,
+// I leave it untouched
+func (tx *Tx) buildBucketInIndex() error {
```
[the remark this answers is not in the document]
Comment (ksankeerth) [reply tx.go:751]: I've updated the branch with changes and fixed broken tests.
When running tests, I've noticed the below tests fails in upstream/master as well(not due to the changes of this PR)
```
--- FAIL: TestDB_MergeForList (37.11s)
    merge_test.go:312:
        	Error Trace:	merge_test.go:312
        	Error:      	Not equal:
        	            	expected: *errors.errorString(&errors.errorString{s:"the number of files waiting to be merged is at least 2"})
        	            	actual  : <nil>(<nil>)
        	Test:       	TestDB_MergeForList
```

Please check when you have time.

[the remark this answers is not in the document]
Comment (bigboss2063) [reply tx.go:749]: I see, thanks.

**Comment by bigboss2063** [comment]
It seems `TestDB_MergeForList` already passed. Is it an error that won't trigger stably?

**Comment by bigboss2063** [comment]
I think the reason for triggering this error is that the folder where this test was conducted was not cleared after the previous test.

**Comment by ksankeerth** [comment]
> I think the reason for triggering this error is that the folder where this test was conducted was not cleared after the previous test.

Yes, That's the reason.  We may change it to a temp directory.

**Comment by bigboss2063** [comment]
Thanks 😁, we will fix it.

**Comment by ksankeerth** [comment]
I've added a few tests to check bucket indexes after bucket creation and deletion.

**Comment by bigboss2063** [comment]
> I've added a few tests to check bucket indexes after bucket creation and deletion.

Thanks. But it seems that you forget combine the methods😁?

**Comment by ksankeerth** [comment]
> > I've added a few tests to check bucket indexes after bucket creation and deletion.
> 
> Thanks. But it seems that you forget combine the methods😁?

Really sorry! I missed them somehow.

**Review by bigboss2063 — APPROVED** [review]
LGTM! Thanks for your great contribution!

**Pull request merged**